scripts/jenkins/ardana/ansible/post-deploy-checks.yml

```diff
@@ -1,134 +1,185 @@
-
-- name: Trim comments and blank lines from rpm added files white-list
-  lineinfile:
-    dest: "{{ rpm_added_files_whitelist }}"
-    regexp: '^(#|$)'
-    state: absent
-
-- name: Check for rogue files introduced by deployment which are not owned by a package
-  shell: |
-    ! find /usr -printf "%p -> %l\n" | \
-      egrep -vf "{{ rpm_added_files_whitelist }}" | \
-      sed 's/ -> .*//' | \
-      xargs -r -d '\n' rpm -qf | \
-      grep -vxFf "{{ pre_deploy_unowned_files }}" | \
-      grep "is not owned"
-  register: rogue_files
-  ignore_errors: true
-
-- name: Record failure if rogue files introduced by deployment
-  set_fact:
-    post_deploy_checks_failures:
-      "{{ post_deploy_checks_failures + [ 'Rogue files introduced by deployment' ] }}"
-  when:
-    - rogue_files | failed
-
-- name: Remove temporary directory if not running in test mode and all checks succeeded
-  become: yes
-  file:
-    path: "{{ verification_temp_dir }}"
-    state: absent
-  when:
-    - rpm_verification_test_mode is not defined
-    - post_deploy_checks_failures|length == 0
+---
+- name: Post deployment checks
+  hosts: hosts
+  gather_facts: False
+
+  vars_files:
+    - ardana_net_vars.yml
+
+  tasks:
+  - name: Initialise array tracking failures
+    set_fact:
+      failures: []
+
+  - name: Configure Cloud for Tempest run
+    shell: |
+      ansible-playbook -v -i hosts/verb_hosts ardana-cloud-configure.yml
+    args:
+      chdir: /var/lib/ardana/scratch/ansible/next/ardana/ansible
+    become: true
+    become_user: ardana
+
+  - name: Run tempest
+    shell: |
+      ansible-playbook -v -i hosts/verb_hosts tempest-run.yml \
+                       -e run_filter="{{ tempest_run_filter }}"
+    args:
+      chdir: /var/lib/ardana/scratch/ansible/next/ardana/ansible
+    become: true
+    become_user: ardana
+    register: tempest_run_result
+    ignore_errors: true
+
+  - name: Record any failure of tempest
+    set_fact:
+      failures: "{{ failures + ['tempest'] }}"
+    when: tempest_run_result|failed
+
+  - name: Set files to contain output of various verification steps
+    set_fact:
+      post_deploy_rpm_verification:
+        "{{ verification_temp_dir }}/rpm-verify-post.out"
+      rpm_verification_fixmes:
+        "{{ verification_temp_dir }}/rpm-verify-modification-whitelist.txt"
+      rpm_modifiable_config_files:
+        "{{ verification_temp_dir }}/rpm-modifiable-config-files.txt"
+      rpm_modifiable_config_file_regexps:
+        "{{ verification_temp_dir }}/rpm-modifiable-config-file-regexps.txt"
+      rpm_added_files_whitelist:
+        "{{ verification_temp_dir }}/rpm-added-files-whitelist.txt"
+
+  - name: Initialise array tracking post-deployment checks failures
+    set_fact:
+      post_deploy_checks_failures: []
+
+  - name: Take snapshot of modified/deleted rpm-owned files post deployment
+    become: yes
+    shell: |
+      rpm -Va | grep -vxFf "{{ pre_deploy_rpm_verification }}" \
```

Comment:
This variable is no longer defined because you are no longer executing the `pre-deploy-checks.yaml` and `post-deploy-checks.yaml` playbooks in the same run.  Maybe worth extracting out the setting of this variable and `pre_deploy_unowned_files` into a separate file which is shared between them.